atlasdb-server/src/main/java/com/palantir/atlasdb/server/config/ClusterConfig.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.server.config;
+
+import java.util.Set;
+
+import javax.validation.constraints.Size;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Preconditions;
+import com.palantir.atlasdb.config.ImmutableLeaderConfig;
+import com.palantir.atlasdb.config.LeaderConfig;
+
+@JsonSerialize(as = ImmutableClusterConfig.class)
+@JsonDeserialize(as = ImmutableClusterConfig.class)
+@Value.Immutable
+public abstract class ClusterConfig {
+    @Value.Default
+    public LeaderElectionConfig leaderElection() {
```

Comment:
Correct. This is a format I went over with the team and the general opinion was that we wanted to move to a config such as this.